test/coverage/datamodel/UuidSerializerTest.cpp

```diff
@@ -23,16 +27,235 @@ class TestFixture : public testing::Test {
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestUuidSerializer() = default;
+	~TestUuidSerializer() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+// Test string serialization
+TEST_F(TestUuidSerializer, SerializeToString) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x1234567890ABCDEF);
+	uuid.set_lsb(0xFEDCBA0987654321);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value
+	EXPECT_EQ(serialized_uuid, "12345678-90ab-cdef-fedc-ba0987654321");
+}
+
+// Test serialization with leading zeros in each segment
+TEST_F(TestUuidSerializer, SerializeWithLeadingZeros) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x00001234007800AB);
+	uuid.set_lsb(0x00FE00BA09876543);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value
+	EXPECT_EQ(serialized_uuid, "00001234-0078-00ab-00fe-00ba09876543");
+}
+
+// Test serialization with mixed case letters
+TEST_F(TestUuidSerializer, SerializeWithMixedCaseLetters) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x1234567890ABcDEF);
+	uuid.set_lsb(0x00dcbA0987654321);
```

Comment:
KUDOS : I don't know if you intentionally mixed the cases here - but GOOD test!

Reply:
Oh you did do it on purpose (just saw the name of the test) -- well comment still applies - good test!